cmd/all-in-one/main.go

```diff
@@ -292,16 +299,7 @@ func startZipkinHTTPAPI(
 	}
 }
 
-func startQuery(
-	qOpts *queryApp.QueryOptions,
-	spanReader spanstore.Reader,
-	depReader dependencystore.Reader,
-	logger *zap.Logger,
-	rootFactory metrics.Factory,
-	baseFactory metrics.Factory,
-	hc *healthcheck.HealthCheck,
-	queryOpts querysvc.QueryServiceOptions,
-) {
+func initTracer(metricsFactory metrics.Factory, logger *zap.Logger) io.Closer {
```

Comment:
This works fine, but more intuitively I think initTracer should return a tracer (instead of closer) which provides a close function.

Reply:
that's not possible given how jaeger-client API is designed (it returns a separate closer).

Reply:
OK